lib/phoenix/channel/server.ex

```diff
@@ -147,9 +147,9 @@ defmodule Phoenix.Channel.Server do
   def reply(pid, ref, topic, {status, payload}, serializer)
       when is_binary(topic) and is_map(payload) do
 
-    send pid, serializer.encode!(
+    send pid, {:channel_push, serializer.encode!(
```

Comment:
The message is already wrapped in the current master, but the wrapping is delegated to the serializer. So [websocket serializer adds a tag](https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/transports/websocket_serializer.ex#L25), while [longpoller uses a struct](https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/transports/long_poll_serializer.ex#L25).

Consequently, [I had to deal with both kinds of messages in the generic socket driver](https://github.com/sasa1977/phoenix/blob/46dcff61caec806373e6dddd2e8fa279c0f6d640/lib/phoenix/channel/driver.ex#L62-L69). Moreover, such driver wouldn't work with other transports, since they might use some other tagging mechanism.

So [I made this change](https://github.com/sasa1977/phoenix/commit/eced974e6069d5bf65dec0e8fd93952c08123867) and made channel server responsible for the tag of the pushed message.